app/models/recipe.rb

```diff
@@ -1,3 +1,9 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
-end
\ No newline at end of file
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def total_cost
+      self.ingredients.pluck(:cost).sum
```

Comment:
Great work! you could also do `ingredients.sum(:cost)` rather than summing after the pluck (which is technically the Ruby version of the sum)